### PR TITLE
Fix #2904 undefined method in chart traker entity

### LINKED ISSUE #2904: Undefined method in Entity ChartTracker
At line [154](https://github.com/zerai/openemr/blob/6e228ee21c6d7fdd6c5b9ffb13fce118593077a4/src/Entities/ChartTracker.php#L154)  there is a call to $this->getDate().
This method is not defined.

### PR DESCRIPTION
**Pull request by zerai**
I have changed the __toString() method, the undefined call "$this->getDate()" is moved to "$this->getWhen()".
I'm not sure about the string format of the date object.

Should be fine I guess.

### CODE REVIEW

File: src/Entities/ChartTracker.php [hunk 1]
```diff
@@ -151,9 +151,9 @@ public function setLocation($value)
     public function __toString()
     {
         return "pid: '" . $this->getPid() . "' " .
-               "date: '" . $this->getDate() . "' " .
+               "date: '" . $this->getWhen()->format('Y-m-m') . "' " .
```
Comment (bradymiller) [inline src/Entities/ChartTracker.php]: Is this supposed to be Y-m-d ?

**Comment by bradymiller** [comment]
nice to see some more testing!!!
(only had one question on the Y-m-m thing)

File: src/Entities/ChartTracker.php (hunk 1 above)
Comment (zerai) [reply src/Entities/ChartTracker.php]: yes... it's a typo..

**Comment by zerai** [comment]
??

**Comment by sjpadgett** [comment]
@zerai Sorry, seems highlight of function failed. I meant the class function __toString().
Suppose it makes better sense seeing it returns all properties so ignore...

**Comment by bradymiller** [comment]
hi @zerai , Bringing this into the codebase. Thanks for the contribution! -brady

**Pull request merged**